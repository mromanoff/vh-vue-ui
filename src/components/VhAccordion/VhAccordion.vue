<template>
  <div class="VHAccordion">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'VhAccordion',
  props: {
    expandMode: {
      type: String,
      default: 'single',
      validator: value => {
        return value.match(/(single|multiple)/);
      },
    },
  },
  mounted() {
    this.$children.forEach(child =>
      child.$on('accordion-item-select', id => {
        this.handleAccordionItemSelect(child, id);
      }),
    );
  },
  methods: {
    handleAccordionItemSelect(child, id) {
      switch (this.expandMode) {
        case 'single':
          this.$children.forEach(child => {
            if (child.id === id) {
              child.isOpen = !child.isOpen;
            } else if (child.id !== id && child.isOpen) {
              child.isOpen = false;
            }
          });
          break;
        case 'multiple':
          child.isOpen = !child.isOpen;
          break;
        default:
          break;
      }
    },
  },
};
</script>

<style scoped>
.VHAccordion {
  position: relative;
}
</style>
