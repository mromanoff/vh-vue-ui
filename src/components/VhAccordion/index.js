import VhAccordion from './VhAccordion.vue';
import VhAccordionItem from './VhAccordionItem.vue';

export { VhAccordion, VhAccordionItem };
