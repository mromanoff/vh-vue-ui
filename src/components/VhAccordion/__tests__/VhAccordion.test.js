import { shallowMount } from '@vue/test-utils';
import VhAccordion from '../VhAccordion.vue';

describe('VhAccordion.vue', () => {
  let wrapper;

  it('renders default accordion', () => {
    wrapper = shallowMount(VhAccordion);
    expect(wrapper.props().expandMode).toBe('single');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders single accordion', () => {
    wrapper = shallowMount(VhAccordion, {
      propsData: {
        expandMode: 'single',
      },
    });
    wrapper.element.setAttribute('expandMode', 'single');
    expect(wrapper.props().expandMode).toBe('single');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders multiple accordion', () => {
    wrapper = shallowMount(VhAccordion, {
      propsData: {
        expandMode: 'multiple',
      },
    });
    wrapper.element.setAttribute('expandMode', 'single');
    expect(wrapper.props().expandMode).toBe('multiple');
    expect(wrapper).toMatchSnapshot();
  });
});
