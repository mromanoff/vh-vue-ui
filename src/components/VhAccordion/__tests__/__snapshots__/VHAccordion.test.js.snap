// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHAccordion.vue renders default accordion 1`] = `<div class="VHAccordion"></div>`;

exports[`VHAccordion.vue renders multiple accordion 1`] = `<div class="VHAccordion" expandmode="single"></div>`;

exports[`VHAccordion.vue renders single accordion 1`] = `<div class="VHAccordion" expandmode="single"></div>`;
